### PR TITLE
Add HMAC-based One Time Password (HOTP) Generator

### PR DESCRIPTION
**Pull request by gaetanomatonti**
### Description
This PR adds the base logic for HMAC-based One-Time Passwords (HOTP) generation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,18 @@
 # Uno
 
-A description of this package.
+A simple OTP generator written in Swift.
+
+## One-Time Password
+An OTP (One-Time Password) is an authentication code that can only be used once. As such, it is often used in combination with regular password to provide an extra layer of security (multi-factor authentication).
+ 
+### HMAC-based One-Time Password
+An HOTP (where the "H" stands for _Hash-based Message Authentication Code_, or _HMAC_) is a one-time password generated from a cryptographic hash function and a static secret key. It also involves a moving factor that seeds into the generator function, usually a counter.
```
Comment (FabrizioBrancati) [inline README.md:9]:
```suggestion
A HOTP (where the "H" stands for _Hash-based Message Authentication Code_, or _HMAC_) is a one-time password generated from a cryptographic hash function and a static secret key. It also involves a moving factor that seeds into the generator function, usually a counter. 
```

File: README.md [hunk 2]
```diff
@@ -1,3 +1,18 @@
 # Uno
 
-A description of this package.
+A simple OTP generator written in Swift.
+
+## One-Time Password
+An OTP (One-Time Password) is an authentication code that can only be used once. As such, it is often used in combination with regular password to provide an extra layer of security (multi-factor authentication).
```
Comment (FabrizioBrancati) [inline README.md:6]:
```suggestion
An OTP (One-Time Password) is an authentication code that can only be used once. As such, it is often used in combination with a regular password to provide an extra layer of security (multi-factor authentication).
```

**Review by FabrizioBrancati — COMMENTED** [review]
I saw that you are pushing the `.swiftpm` folder, usually this folder is added into the `.gitignore` file.
Also, you should use a template for your `.gitignore` file, [something like that](https://www.toptal.com/developers/gitignore/api/swiftpm,code), for example.

**Comment by gaetanomatonti** [comment]
> I saw that you are pushing the `.swiftpm` folder, usually this folder is added into the `.gitignore` file.
> Also, you should use a template for your `.gitignore` file, [something like that](https://www.toptal.com/developers/gitignore/api/swiftpm,code), for example.

@FabrizioBrancati Thanks for the suggestion! As for the gitignore it is already ignoring the folders and files in the template you suggested – as it is the default `.gitignore` created by Xcode. I'll just ignore and remove the whole `.swiftpm` folder.

**Comment by FabrizioBrancati** [comment]
My suggestion was meant for maintainability because by using a template, with an update link inside, you will be able to keep it easily updated.
Anyway, it's just a suggestion 😉

**Review by FabrizioBrancati — APPROVED** [review]
LGTM 👍

**Comment by gaetanomatonti** [comment]
> My suggestion was meant for maintainability because by using a template, with an update link inside, you will be able to keep it easily updated.
> Anyway, it's just a suggestion 😉

Ah I see now! I'll use the template then.

**Pull request merged**